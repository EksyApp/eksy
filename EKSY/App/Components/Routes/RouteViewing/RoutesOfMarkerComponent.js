import React, {Component} from 'react'
import Container from '../../Common/Container'
import RouteCardList from './RouteCardList'
import {StyleSheet} from 'react-native'
import Message from '../../Common/Message'
import PropTypes from 'prop-types'
import {RoutesShape} from "../../../Utils/PropTypeShapes";

export default class RoutesOfMarkerComponent extends Component {
	
	renderList() {
		if(this.props.loading) {
			return(<Message>Loading...</Message>)
		} else {
			return(<RouteCardList routes={this.props.routes} onPress={(route) => this.props.onCardClick(route)} style={styles.card}/>)
		}
	}
	
	render() {
		return(
				<Container backButton title="Start a route">
					{this.renderList()}
				</Container>
		)
	}
	
}

const styles = StyleSheet.create({
	card: {
		width: '95%',
		alignSelf: 'center'
	}
})

RoutesOfMarkerComponent.propTypes = {
	loading: PropTypes.bool,
	routes: RoutesShape,
	onCardClick: PropTypes.func
}