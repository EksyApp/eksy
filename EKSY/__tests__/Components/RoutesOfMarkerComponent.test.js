import React from 'react';
import renderer from 'react-test-renderer';
import RoutesOfMarkerComponent from "../../App/Components/Routes/RouteViewing/RoutesOfMarkerComponent";

test('RouteViewComponent renders correctly', () => {
	const tree = renderer.create(
			<RoutesOfMarkerComponent
					routes = {[{markers: [{latitude: 1, longitude: 1}]}]}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});