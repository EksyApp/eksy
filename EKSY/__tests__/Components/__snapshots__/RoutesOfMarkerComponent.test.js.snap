// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteViewComponent renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(243,244,229)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(226,227,217)",
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            Object {
              "height": 50,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 50,
              "zIndex": 1,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 35,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgb(101,114,122)",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î—„
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(101,114,122)",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Start a route
    </Text>
  </View>
  <RCTScrollView
    backButton={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgb(243,244,229)",
          "flex": 1,
        },
        undefined,
      ]
    }
    title="Start a route"
  >
    <View>
      <RCTScrollView
        dataSource={
          ListViewDataSource {
            "items": 1,
          }
        }
        enableEmptySections={true}
        renderRow={[Function]}
        renderScrollComponent={[Function]}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(226,227,217)",
                  "borderBottomWidth": 0,
                  "borderColor": "#ddd",
                  "borderRadius": 2,
                  "borderWidth": 1,
                  "elevation": 1,
                  "marginLeft": 5,
                  "marginTop": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 2,
                },
                Array [
                  Object {
                    "height": 200,
                    "marginLeft": 0,
                    "width": "100%",
                  },
                  Object {
                    "width": undefined,
                  },
                  Object {
                    "alignSelf": "center",
                    "width": "95%",
                  },
                ],
              ]
            }
            testID={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(112,112,112)",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "padding": 5,
                    "position": "relative",
                  },
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "overflow": "scroll",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "rgb(101,114,122)",
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    Object {
                      "marginTop": 10,
                    },
                  ]
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "rgb(89,85,78)",
                    "fontSize": 18,
                    "margin": 10,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
